rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -806,20 +840,118 @@ public License convertLicenseFromRequest(Map<String, Object> reqBodyMap, License
+                          fieldName, licenseId);
             }
         }
+
         if (!licenseIncorrect.isEmpty()) {
-            throw new BadRequestClientException("License with ids " + licenseIncorrect + " does not exist in SW360 database.");
+            String errorMsg = fieldName + " with ids " + licenseIncorrect + " does not exist in SW360 database.";
+            if (!licenseWarnings.isEmpty()) {
+                errorMsg += " Note: External license references " + licenseWarnings + " were allowed.";
+            }
+            LOGGER.error("License validation failed: {}", errorMsg);
+            throw new BadRequestClientException(errorMsg);
         }
+
+        if (!licenseWarnings.isEmpty()) {
+            LOGGER.info("{} update proceeding with external license references: {}", fieldName, licenseWarnings);
+        }
+    }
+
+    /**
+     * Helper method to determine if a license ID appears to be an external reference
+     * (e.g., long descriptive names, proprietary licenses) rather than an internal SW360 ID
+     * This method is conservative to avoid false positives with common SPDX identifiers
+     */
+    private boolean isLikelyExternalLicenseReference(String licenseId) {
+        if (licenseId == null || licenseId.trim().isEmpty()) {
+            return false;
+        }
+
+        String normalizedId = licenseId.toLowerCase().trim();
+
+        return isMatchingConfigurablePatterns(normalizedId) ||
+               isLongDescriptiveName(normalizedId);
+    }
+
+    /**
+     * Check against configurable license patterns from application properties
+     * This is now more conservative to avoid false positives with standard SPDX identifiers
+     */
+    private boolean isMatchingConfigurablePatterns(String normalizedId) {
+
+        if ((normalizedId.endsWith("license") || normalizedId.endsWith("licence")) && 
+            normalizedId.length() > 10) {
+            return true;
+        }
+
+        if ((normalizedId.startsWith(".") || normalizedId.contains("microsoft") || 
+             normalizedId.contains("oracle") || normalizedId.contains("google") ||
+             normalizedId.contains("proprietary") || normalizedId.contains("commercial")) && 
+            normalizedId.contains("license")) {
+            return true;
+        }
+
+        if (normalizedId.startsWith("gnu ") && normalizedId.contains("general public license")) {
+            return true;
+        }
+
+        if (normalizedId.startsWith("creative commons") || normalizedId.contains("cc-by")) {
+            return true;
+        }
+
+        if (normalizedId.matches(".*\\bversion\\s+\\d+.*") && normalizedId.length() > 15) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if it's likely a long descriptive name from external sources
+     * Only very long names are considered external to be conservative
+     */
+    private boolean isLongDescriptiveName(String normalizedId) {
```

Comment:
Same with this function. Why are we looking for arbitrary value of `40` in name lenght?